core/templates/dev/head/collection_player/CollectionDataService.js

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to retrieve information about collections from the
+ * backend.
+ *
+ * @author henning.benmax@gmail.com (Ben Henning)
+ */
+
+// TODO(bhenning): For preview mode, this service should be omitted. Instead,
+// a separate collectionDataService should be created which serves a local copy
+// of the collection instead.
+oppia.factory('CollectionDataService', [
+    '$http', 'COLLECTION_DATA_URL', 'UrlInterpolationService',
+    function($http, COLLECTION_DATA_URL, urlInterpolationService) {
+
+  // Maps previously loaded collections to their IDs.
+  var _collectionCache = [];
+
+  // This inserts an extra 'exploration' object into each node. The object
+  // includes the exploration ID and additional meta-information regarding the
+  // exploration that is useful for collection pages (such as the exploration
+  // title).
+  var _populateExplorationsInNodes = function(collection, explorationTitles) {
+    for (var i = 0; i < collection.nodes.length; i++) {
+      var collectionNode = collection.nodes[i];
+      collectionNode.exploration = {
+        'id': collectionNode.exploration_id,
+        'title': explorationTitles[collectionNode.exploration_id]
+      };
+    }
+    return collection;
+  };
+
+  // TODO(bhenning): Figure out what to do about versioning. Should the version
```

Comment:
I'm inclined not to support versioning and just use latest. I thought we talked about this in the design phase, though; what conclusion did we end up with?

Might be something to discuss with Xiaotong, too.